src/state/page-models/actions.js

```diff
@@ -143,3 +159,27 @@ export const createPageModel = pageModel => (dispatch, getState) => {
     }
   });
 };
+
+const fetchCurrentReference = (getApiCall, setActionCreator) =>
+  (page = { page: 1, pageSize: 10 }) => (dispatch, getState) =>
+    new Promise((resolve) => {
+      const { pageModelCode } = getParams(getState());
+      dispatch(toggleLoading('references'));
+      getApiCall(pageModelCode, page).then((response) => {
+        response.json().then((json) => {
+          if (response.ok) {
+            dispatch(setActionCreator(json.payload));
+            dispatch(toggleLoading('references'));
+            dispatch(setPage(json.metaData));
+            resolve();
+          } else {
+            dispatch(addErrors(json.errors.map(err => err.message)));
+            dispatch(toggleLoading('references'));
+            resolve();
+          }
+        });
+      });
+    });
```

Comment:
you can optimize the function like this: 

const fetchCurrentReference = (getApiCall, setActionCreator) =>
  (page = { page: 1, pageSize: 10 }) => (dispatch, getState) =>
    new Promise((resolve) => {
      const { pageModelCode } = getParams(getState());
      dispatch(toggleLoading('references'));
      getApiCall(pageModelCode, page).then((response) => {
        response.json().then((json) => {
          if (response.ok) {
            dispatch(setActionCreator(json.payload));
            dispatch(setPage(json.metaData));
          } else {
            dispatch(addErrors(json.errors.map(err => err.message)));
          }
            dispatch(toggleLoading('references'));
            resolve();
        });
      });
    });